### PR TITLE
removed cat, equal, and stack from autocast promote list

### PR DESCRIPTION
**Pull request by Fuzzkatt**
Fixes #{issue number}

### CODE REVIEW

**Review by mcarilli — APPROVED** [review]

**Comment by mcarilli** [comment]
Looks good to me, but i'm not the final authority.

**Comment by ngimel** [comment]
Is it because you can rely on cat/equal/stack internal type promotion to handle different type inputs?
Can you please rebase to get CI signal?

**Comment by mcarilli** [comment]
> Is it because you can rely on cat/equal/stack internal type promotion to handle different type inputs?

Yes. When I originally wrote the autocast lists cat/equal/stack didn't have internal type promotion. Part of Patrick's project is to overhaul the cast lists by checking new op capabilities, and now those three apparently do handle mixed input dtypes with internal promotion, so the PR moves them off the promote list.

The PR also moves their tests onto the test-case list `torch_expect_builtin_promote` which does what it says on the tin: tries mixed input dtypes without autocast, to make sure our internal-promotion expectation doesn't break over time.

**Review by ngimel — APPROVED** [review]

**Comment by ngimel** [comment]
Lint error is real

File: torch/testing/_internal/autocast_test_lists.py [hunk 1]
```diff
@@ -61,6 +61,10 @@ def __init__(self, dev):
             ("add", pointwise0_fp32 + pointwise1_fp16, torch.float32),
             ("div", pointwise0_fp32 + pointwise1_fp16, torch.float32),
             ("mul", pointwise0_fp32 + pointwise1_fp16, torch.float32),
+            ("cat", (pointwise0_fp16 + pointwise1_fp32,), torch.float32),
+            ("equal", pointwise0_fp32 + pointwise1_fp16, torch.float32),
+            ("stack", (pointwise0_fp16 + pointwise1_fp32,), torch.float32),
+
```
Comment (ngimel) [inline torch/testing/_internal/autocast_test_lists.py:67]:
```suggestion

```

**Pull request closed without merging**